.buildkite/pipeline.yml

```diff
@@ -0,0 +1,41 @@
+common-params:
+  &publish-android-artifacts-docker-container
+  docker#v3.8.0:
+    image: "public.ecr.aws/automattic/android-build-image:v1.1.0"
+    propagate-environment: true
+    environment:
+      # DO NOT MANUALLY SET THESE VALUES!
+      # They are passed from the Buildkite agent to the Docker container
+      - "AWS_ACCESS_KEY"
+      - "AWS_SECRET_KEY"
+
+steps:
+  - label: "Lint & Checkstyle"
+    key: "lint_and_checkstyle"
+    plugins:
+      - *publish-android-artifacts-docker-container
+    command: |
+      cp gradle.properties-example gradle.properties
+      ./gradlew lint checkstyle
+    artifact_paths:
+      - "**/build/reports/lint-results.*"
+      - "**/build/reports/checkstyle/checkstyle.*"
+  - label: "Test"
+    key: "test"
+    plugins:
+      - *publish-android-artifacts-docker-container
+    command: |
+      cp gradle.properties-example gradle.properties
+      ./gradlew test
+  # Publish tasks
+  - label: "Publish to s3"
+    depends_on:
+      - "lint_and_checkstyle"
+      - "test"
+    plugins:
+      - *publish-android-artifacts-docker-container
+    command: |
+      cp gradle.properties-example gradle.properties
+      ./gradlew \
+          :WordPressLoginFlow:prepareToPublishToS3 $(prepare_to_publish_to_s3_params) \
+          :WordPressLoginFlow:publish
```

Comment:
Note to self (and any future reviewer):

 - I notice that CircleCI config had the `filters: { tags: { only: /.*/ } }` so at first I incorrectly thought that this meant that the publish job was _only_ run on tags, but not on regular pushes on branches. Which would mean we'd have a difference in this Buildkite config compared to that old setup.
 - But my assumption was wrong, because in fact [CircleCI will _always_ build on all branches (unless you specify an explicit `filters: { branches: … }`)](https://circleci.com/docs/2.0/configuration-reference/#branches) so, contradictory to intuition, that syntax in the old CircleCI config meant "also build on tags only matching `.*` (aka all tags)… [in addition to building for all branches like it will do by default"

So TL;DR: that confirms that despite the apparences, the new Buildkite config for this step _does_ match the behavior we had in the old CircleCI config for that job 👌